ovhdata-common/src/ovhapi/di_api.rs

```diff
@@ -259,6 +289,26 @@ impl DiApi for OVHapiV6Client {
         response.parse().await
     }
 
+    async fn di_destinations_filtered(&self, service_name: &str, filter: Option<String>) -> Result<Vec<Destination>> {
+        let request = self
+            .build_request(
+                Method::GET,
+                &["cloud", "project", service_name, "dataIntegration", "destinations"],
+                &[],
+                &HeaderMap::new(),
+                EMPTY_BODY,
+            )
+            .await?;
+        let response = request.send(&self.client, &[]).await?;
+
+        // Filter the list before parsing it with jsonpath filter
+        let body_string = response.body_text().await;
+        let json : Value = serde_json::from_str(body_string.as_str()).map_err(|e| Error::DeserializeContent(e, body_string))?;
+
+        let new_response = jsonpath::json_parse_task(json, filter).map_err(Error::FilterContent)?;
```

Comment:
Could be integrated generically into the ResponseWrapper via an optional argument of the parse method